linkerd/app/src/env.rs

```diff
@@ -537,18 +537,36 @@ pub fn parse_config<S: Strings>(strings: &S) -> Result<super::Config, EnvError>
             }
 
             let default = parse(strings, ENV_INBOUND_DEFAULT_POLICY, |s| {
-                s.parse().map_err(ParseError::from)
+                parse_default_policy(s, detect_protocol_timeout)
             })?
-            .unwrap_or_default();
-
-            const ALLOW_OPAQUE: inbound::AllowPolicy =
-                inbound::AllowPolicy::Unauthenticated { skip_detect: true };
+            .unwrap_or_else(|| {
+                parse_default_policy("unauthenticated", detect_protocol_timeout)
+                    .expect("unauthenticated must be a valid default policy")
```

Comment:
not a big deal, but it seems a bit silly to try to parse this from a string? I'm guessing that the intent was to show _what_ policy was the default here, instead of just calling `Default::default()`...but maybe just having something like
```rust
.unwrap_or_else(|| DefaultPolicy::unauthenticated(detect_protocol_timeout))
```
would feel a little less circuitous?

not a blocker